src/Serializers/QueryResultSerializer.php

```diff
@@ -67,13 +67,39 @@ public function isSerializerFor( $queryResult ) {
 	public static function getSerialization( DataItem $dataItem, $printRequest = null ) {
 		switch ( $dataItem->getDIType() ) {
 			case DataItem::TYPE_WIKIPAGE:
-				$title = $dataItem->getTitle();
-				$result = array(
-					'fulltext' => $title->getFullText(),
-					'fullurl' => $title->getFullUrl(),
-					'namespace' => $title->getNamespace(),
-					'exists' => $title->isKnown()
-				);
+
+				// Support for a deserializable _rec type with 0.6
+				if ( $printRequest !== null && $printRequest->getTypeID() === '_rec' ) {
+					$recordValue = DataValueFactory::getInstance()->newDataItemValue(
+						$dataItem,
+						$printRequest->getData()->getDataItem()
+					);
+
+					$recordDiValues = array();
+
+					foreach ( $recordValue->getPropertyDataItems() as $property ) {
+						$label = $property->getLabel();
+
+						$recordDiValues[$label] = array(
+							'label'  => $label,
+							'typeid' => $property->findPropertyTypeID(),
+							'item'   => array()
+						);
+
+						foreach ( $recordValue->getDataItem()->getSemanticData()->getPropertyValues( $property ) as $value ) {
+							$recordDiValues[$label]['item'][] = self::getSerialization( $value );
+						}
+					}
+					$result = $recordDiValues;
+				} else {
+					$title = $dataItem->getTitle();
+					$result = array(
+						'fulltext' => $title->getFullText(),
+						'fullurl' => $title->getFullUrl(),
+						'namespace' => $title->getNamespace(),
+						'exists' => $title->isKnown()
+					);
+				}
```

Comment:
I know but no it can't because the method is static and I'm not going to fix legacy code here for the sake of cleanliness. This has to be done in a follow-up.

This PR is here to fix a problem to existed for years which only came to light by trying to deserialze a _rec DV on the other end of a API response.

I have too much other stuff on my list to engage in fixing the `QueryResultSerializer` serialization process.